TASK: Remove old UI specific content wrappings and CSS

markusguenther opened the pull request:
The nodetype schema endpoint saves all values with the prefix typo3. This change will replace this with neoscms.

**What I did**

- [x] Removed all content wrappings related to the old Neos UI
- [x] Removed PHP code that is completely replaced via AOP by Neos UI code

**How to verify it**

Every content element when rendered in the backend should have much less data attributes, especially the `typo3` prefixed ones.
All should still work the same, but cleaner and faster.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

markusguenther committed:
TASK: Replace typo3 prefix in the schema with neoscms

The nodetype schema endpoint saves all values with the prefix typo3. This change will replace this with neoscms.

markusguenther commented:
Not sure if we have a namespace URL for neos.io, so that I can also replace the typo3.org URL. But the URL is not valid anymore.

mhsdesign commented:
needs the neos ui to be adjusted? or doesnt rely it on the attributes `property="typo3:something"`

Sebobo commented:
I think you have to rebase as I removed all the VIE stuff.

kdambekalns commented:
- URL used as namespace identifiers do not have to exist or point to something "useful". So you can "invent" something, like `neos.ui/ns/…`
- Can this go in 8.0, still?

markusguenther committed:
TASK: Adjust URL to neos.io

markusguenther committed:
TASK: Remove vieSchemaBuilder

markusguenther commented:
As the vieSchema is not needed anymore this PR removes the test and the Service.
https://github.com/neos/neos-development-collection/pull/3686

kdambekalns reviewed (approved):
I dare to approve by reading… 🙈

Sebobo commented:
I honestly think we can remove a ton of code around that area. The UI package adds its own data attributes via aspects to the elements. The ones that are still added by the Neos classes are almost all unnecessary. I removed most and editing etc. still worked fine.

Question: Should we try to clean that up now or in 8.1?
Shouldn't be breaking so we don't need to force it IMO.

Sebobo commented:
I honestly think we can remove a ton of code around that area. The UI package adds its own data attributes via aspects to the elements. The ones that are still added by the Neos classes are almost all unnecessary. I removed most and editing etc. still worked fine.

Question: Should we try to clean that up now or in 8.1?
Shouldn't be breaking so we don't need to force it IMO.

Sebobo committed:
TASK: Remove deprecated content element attributes

Neos.Ui adds their own properties necessary for inline editing.
The removed ones were only used for the old aloha editor.

The ContentElementEditableService is now only used
for extensibility f.e. in the Neos.Ui package.

Sebobo commented:
I removed more stuff I found in https://github.com/markusguenther/neos-development-collection/pull/2
Tests are adjusted and green. So we can decide whether we want to do the real cleanup of the old stuff.

mhsdesign commented:
@Sebobo the ui aop overridea the editable/content wrapping service see https://github.com/neos/neos-development-collection/issues/3595

Sebobo commented:
@mhsdesign yes I saw that. So your issue would be a follow up to make this more api like right?

mhsdesign commented:
yes to clean this up in some way but its hard to get right (as it would be also wierd to have some random 'services' laying in the neos ui which will be configured in Objects.yaml so they are injected in Neos.Neos)

markusguenther committed:
Merge pull request #2 from Sebobo/task/remove-deprecated-node-attributes

TASK: Remove deprecated content element attributes

Sebobo commented:
So my removal of a deprecated stuff is now merged. @mhsdesign maybe you can have a look.
Then we should rename the change.

mhsdesign commented on Neos.Neos/Classes/Service/ContentElementWrappingService.php:
```diff
@@ -47,124 +46,45 @@ class ContentElementWrappingService
-     * @return string
      */
-    public function wrapContentObject(NodeInterface $node, $content, $fusionPath, array $additionalAttributes = [])
+    public function wrapContentObject(NodeInterface $node, string $content, string $fusionPath, array $additionalAttributes = []): string
```
we dont need this method aswell:

https://github.com/neos/neos-ui/blob/e53f8e20ee70ca832828f033f5a69c8223a2deab/Classes/Aspects/AugmentationAspect.php#L103-L112

its completely aop'd (without `$joinPoint->getAdviceChain()->proceed($joinPoint);`)

Sebobo replied:
Thx done

mhsdesign commented on Neos.Neos/Classes/Service/ContentElementEditableService.php:
```diff
@@ -27,50 +23,12 @@
+     * This method is extended by the Neos.Ui package via an aspect to add the needed markup for inline editing.
      */
-    public function wrapContentProperty(NodeInterface $node, $property, $content)
+    public function wrapContentProperty(NodeInterface $node, string $property, string $content): string
```
yes this is indeed aop'd via:
https://github.com/neos/neos-ui/blob/e53f8e20ee70ca832828f033f5a69c8223a2deab/Classes/Aspects/AugmentationAspect.php#L142-L150

but `$inBackend` we use `$content = $joinPoint->getAdviceChain()->proceed($joinPoint);` instead of `$content = $joinPoint->getMethodArgument('content')` 

this seems odd we basically use `htmlAugmenter->addAttributes` twice - so removing the code here is not 100% the same, as we will only use `htmlAugmenter->addAttributes` once -> it might still work though.

Sebobo replied:
Thx done

mhsdesign replied:
from seb

> Die 2 Attribute wurden nur vom Styling und Tests verwendet
> Der Rest war dupliziert

mhsdesign reviewed (changes requested):
Thanks for taking care - this is the right step to remove things that are aop'd anyway so we can refactor soon better ;)

Sebobo committed:
Merge branch '8.2' into task/replace-typo3-in-schema

Sebobo committed:
TASK: Removed all unused attributes in the content augmentation

All relevenat augmentation is done in the Neos.Ui package

Sebobo committed:
TASK: Remove unused old UI CSS

mhsdesign reviewed (approved):
looks fine by reading

ill try to check this out locally too ^^

kdambekalns commented:
Ah, I can haz running tests…

Sebobo commented:
Green green green ... 🎶

Sebobo commented:
@mhsdesign did you check again if it breaks the UI?

I would like to merge this one

markusguenther committed:
Merge remote-tracking branch 'composer/8.2' into task/replace-typo3-in-schema

# Conflicts:
#	Neos.Neos/Resources/Public/Styles/Error.css.map

markusguenther committed:
TASK: Add updated yarn lock file

markusguenther commented:
Resolved conflicts and hopefully after that also everything is green :D
